lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -133,19 +133,110 @@ class StableHLOToTTIRTransposeOpConversionPattern
     tensor::EmptyOp outputTensor = rewriter.create<tensor::EmptyOp>(
         srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
 
-    assert(adaptor.getPermutation().size() == 2 &&
-           "TTIR only supports only two dimensional transposeOp.");
+    if (!checkBasicLegality(srcOp, adaptor)) {
+      LLVM_DEBUG(llvm::dbgs() << "Failed legality checks in the handling of Op "
+                              << srcOp->getName());
```

Comment:
Maybe I'm missing something but what's the point of this? We will already throw the error with the specific message what failed.

Reply:
The purpose of this is to start documenting the flow of the program execution in a log file. I just setup DEBUG_TYPE so we can all start logging that aids in debugging.

Reply:
Isn't it better then to catch the errors in one place somewhere top level and to just log their message there? This seems kind of redundant.

Reply:
Ok I should have explained it in detail, my purpose of LLVM_DEBUG() at this stage was NOT to log errors but instead to inform about the flow of the program execution to aid in debugging e.g. we should have LLVM_DEBUG() at each entry and exit points of functions so developer can tell the flow of execution just by looking at the logs rather than having to gdb through the code every time. 

But I should have introduced this logging in separate PR so not to bundle it up with error checking, so I am removing it and will open a separate PR for it.

@nsmithtt Sure I will replace it.

Reply:
That sounds okay, though you should probably check with rest of uhuru team about that (if you already haven't) to make sure the whole project is using the same logging infrastructure and to agree on what tools/infra we want to use for that.

Reply:
@uazizTT, rest of review looks good to me, but perhaps we can use `rewriter.notifyMatchFailure` here before merging?

Reply:
Updated to use `rewriter.notifyMatchFailure`

Reply:
Just to make sure we are all in clear for the future changes, @nsmithtt do you think it is okay that we are using `srcOp->emitError()` inside `checkBasicLegality` function and then also using `rewriter.notifyMatchFailure` in case `checkBasicLegality` returns false, or do you think it would be better if we change this to something like:
```
LogicalResult legalityResult = checkBasicLegality(...);
if (!legalityResult.succeeded()) {
  return legalityResult;
}

...

LogicalResult checkBasicLegality(...) const {
  if (...) {
    return rewriter.notifyMatchFailure(srcOp, "error message");
  }

  return success();
}
```

Latter seems a bit cleaner to me.

Reply:
Yeah I think we should use one API, let me update it for all conversions.

Reply:
I agree, latter is cleaner to me too. @uazizTT, sorry for bikeshedding your review, but since this is fresh code I think it's good to establish the way we want to organize this kind of stuff.  In the future this code will definitely be copy-pasted so it's good to establish best practices at the start.

Reply:
> @uazizTT, sorry for bikeshedding your review, but since this is fresh code I think it's good to establish the way we want to organize this kind of stuff. In the future this code will definitely be copy-pasted so it's good to establish best practices at the start.

+1 I wanted to say the same thing :)

Reply:
OK I refactored and push the changes.